sql/hive/src/test/scala/org/apache/spark/sql/hive/client/VersionsSuite.scala

```diff
@@ -170,6 +171,34 @@ class VersionsSuite extends SparkFunSuite with Logging {
       client.createDatabase(tempDB, ignoreIfExists = true)
     }
 
+    test(s"$version: create/get/alter database should pick right user name as owner") {
+      if (version != "0.12") {
+        val currentUser = UserGroupInformation.getCurrentUser.getUserName
+        val ownerName = "SPARK_29425"
+        val db1 = "SPARK_29425_1"
+        val db2 = "SPARK_29425_2"
+        val ownerProps = Map("ownerName" -> ownerName)
+
+        // create database with owner
+        val dbWithOwner = CatalogDatabase(db1, "desc", Utils.createTempDir().toURI, ownerProps)
+        client.createDatabase(dbWithOwner, ignoreIfExists = true)
+        val getDbWithOwner = client.getDatabase(db1)
+        assert(getDbWithOwner.properties("ownerName") === ownerName)
+        // alter database without owner
+        client.alterDatabase(getDbWithOwner.copy(properties = Map()))
+        assert(client.getDatabase(getDbWithOwner.name).properties("ownerName") === currentUser)
```

Comment:
nit: `getDbWithOwner.name` -> `db1`

Reply:
let's also add a comment that right now `alterDatabase` will reset the owner name to current user

Reply:
hmm I'm confused.  https://github.com/apache/spark/pull/26080/files#diff-b7094baa12601424a5d19cb930e3402fR410 shows that `alterDatabase` can retain the owner info?

Reply:
If we just alter databases with non-ownership props, we should respect the original owner not try to fix it with our spark user.
If we alter database with ownerName, we then change it, this will be replaced by `SET OWNER`